modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -494,25 +495,27 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
 
   if (Moose::FV::onBoundary(*this, fi))
   {
-    const auto sub_id =
-        hasBlocks(elem->subdomain_id()) ? elem->subdomain_id() : neighbor->subdomain_id();
-    const Moose::SingleSidedFaceArg boundary_face{
-        &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, sub_id};
+    const Elem * const boundary_elem = hasBlocks(elem->subdomain_id()) ? elem : neighbor;
+    const Moose::FaceArg boundary_face{
+        &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, boundary_elem};
     return vel(boundary_face);
   }
 
   VectorValue<ADReal> velocity;
 
+  Moose::FaceArg face{
+      &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, nullptr};
   // Create the average face velocity (not corrected using RhieChow yet)
-  velocity(0) = u->getInternalFaceValue(fi, correct_skewness);
+  velocity(0) = (*u)(face);
   if (v)
-    velocity(1) = v->getInternalFaceValue(fi, correct_skewness);
+    velocity(1) = (*v)(face);
   if (w)
-    velocity(2) = w->getInternalFaceValue(fi, correct_skewness);
+    velocity(2) = (*w)(face);
 
-  // Return if Rhie-Chow was not requested
-  if (m == Moose::FV::InterpMethod::Average)
+  // Return if Rhie-Chow was not requested or if we have a porosity jump
+  if (m == Moose::FV::InterpMethod::Average || std::get<0>(epsilon(tid).isDiscontinuous(fi)))
```

Comment:
I am more and more convinced that we should separate the check and evaluation functionalities.

Reply:
nah